lib/spack/spack/package_base.py

```diff
@@ -207,17 +207,24 @@ def __init__(cls, name, bases, attr_dict):
         # If a package has the executables or libraries  attribute then it's
         # assumed to be detectable
         if hasattr(cls, 'executables') or hasattr(cls, 'libraries'):
-            @property
-            def platform_executables(self):
+            @classmethod
+            def platform_executables(cls):
                 def to_windows_exe(exe):
                     if exe.endswith('$'):
                         exe = exe.replace('$', '%s$' % spack.util.path.win_exe_ext())
                     else:
                         exe += spack.util.path.win_exe_ext()
                     return exe
                 plat_exe = []
-                if hasattr(self, 'executables'):
-                    for exe in self.executables:
+                if hasattr(cls, 'executables'):
+                    # Executable can be a standalone class attribute or
+                    # a class method in more complex cases
+                    if callable(cls.executables):
+                        exes = cls.executables()
```

Comment:
can't we make these `classproperty` when needed?

Reply:
Oh, sure, this is a relic of a previous implementation that I didn't remove completely. There should be no change here, thanks for spotting.

Reply:
Fixed in 019e8f10d0755